lib/more_core_extensions/core_ext/object/descendants.rb

```diff
@@ -0,0 +1,17 @@
+require 'active_support/core_ext/class/subclasses'
+
+module MoreCoreExtensions
+  module Descendants
+    #
+    # Retrieve a descendant by its name
+    #
+    def descendant_get(desc_name)
+      return self if desc_name == to_s || desc_name.nil?
+      klass = descendants.find { |desc| desc.name == desc_name }
+      raise "#{desc_name} is not a descendant of #{name}" unless klass
```

Comment:
What do you think of raising `NameError` here?  I think the message itself is great.